omod/src/main/webapp/resources/scripts/patientsearch/patientSearchWidget.js

```diff
@@ -804,4 +809,27 @@ function PatientSearchWidget(configuration){
         // For some reason without this the cursor is at position 0 of the input, i.e. before the initial value
         input[0].selectionStart = input.val().length;
     }
+
+
+    // Store the original XMLHttpRequest open function to preserve its functionality
+    const originalXHR = window.XMLHttpRequest.prototype.open;
+
+    // Override the XMLHttpRequest open function to intercept specific requests
+    window.XMLHttpRequest.prototype.open = function (method, url) {
+
+        // Check if the requested URL contains '/openmrs/registrationapp/biometrics/biometrics/search.action'
+        if (url.includes('/openmrs/registrationapp/biometrics/biometrics/search.action')) {
+
+            jq('#patient-search-form').after(spinnerImage);
+
+            // Apply CSS styles to position the spinner
```

Comment:
This seems a little too specific to one particular form where this widget is used.  This widget is potentially used in a lot of different places, and it may not have the same positioning on the page, etc.  But I need to refresh myself on how all of this is wired together.

Reply:
I agree with you @mseaton . I've removed the css properties.